File: doc-examples/jdbc-example-java/src/main/java/example/ManufacturerRepository.java

package example;

import io.micronaut.context.annotation.Requires;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.GenericRepository;

@JdbcRepository(dialect = Dialect.H2)
@Requires(notEnv="oracle")
public interface ManufacturerRepository extends GenericRepository<Manufacturer, Long> {
    Manufacturer findByName(String name);

    Manufacturer save(String name);

    void deleteAll();
}

File: doc-examples/jdbc-example-python/src/main/python/example/ManufacturerRepository.py
from micronaut.data.jdbc.annotation import JdbcRepository
from micronaut.data.model.query.builder.sql import Dialect
from micronaut.data.repository import GenericRepository

from example.Manufacturer import Manufacturer


@JdbcRepository(dialect=Dialect.H2)
class ManufacturerRepository(GenericRepository[Manufacturer, int]):
    def findByName(self, name: str) -> Manufacturer: ...

    def save(self, name: str) -> Manufacturer: ...

    def deleteAll(self) -> None: ...
